packages/kbn-unified-data-table/src/components/compare_documents/hooks/use_comparison_columns.tsx

```diff
@@ -0,0 +1,147 @@
+
+export interface UseComparisonColumnsProps {
+  wrapper: HTMLElement | null;
+  isPlainRecord: boolean;
+  fieldColumnId: string;
+  selectedDocs: string[];
+  getDocById: (docId: string) => DataTableRecord | undefined;
+  setSelectedDocs: (selectedDocs: string[]) => void;
+}
+
+export const DEFAULT_COLUMN_WIDTH = 300;
+export const FIELD_COLUMN_WIDTH = 200;
+export const FIELD_COLUMN_NAME = i18n.translate('unifiedDataTable.fieldColumnTitle', {
+  defaultMessage: 'Field',
+});
+
+export const useComparisonColumns = ({
+  wrapper,
+  isPlainRecord,
+  fieldColumnId,
+  selectedDocs,
+  getDocById,
+  setSelectedDocs,
+}: UseComparisonColumnsProps) => {
+  const comparisonColumns = useMemo<EuiDataGridColumn[]>(() => {
+    const fieldsColumn: EuiDataGridColumn = {
+      id: fieldColumnId,
+      displayAsText: FIELD_COLUMN_NAME,
+      initialWidth: FIELD_COLUMN_WIDTH,
+      isSortable: false,
+      isExpandable: false,
+      actions: false,
+    };
+
+    const currentColumns = [fieldsColumn];
+    const wrapperWidth = wrapper?.offsetWidth ?? 0;
+    const columnWidth =
+      DEFAULT_COLUMN_WIDTH * selectedDocs.length + FIELD_COLUMN_WIDTH > wrapperWidth
+        ? DEFAULT_COLUMN_WIDTH
+        : undefined;
+
+    selectedDocs.forEach((docId, docIndex) => {
+      const doc = getDocById(docId);
+
+      if (!doc) {
+        return;
+      }
+
+      const additional: EuiListGroupItemProps[] = [];
+
+      if (docIndex !== 0) {
+        additional.push({
+          iconType: 'pin',
+          label: i18n.translate('unifiedDataTable.pinForComparison', {
+            defaultMessage: 'Pin for comparison',
+          }),
+          size: 'xs',
+          onClick: () => {
+            const newSelectedDocs = [...selectedDocs];
+            const index = newSelectedDocs.indexOf(docId);
+            const [baseDocId] = newSelectedDocs;
+
+            newSelectedDocs[0] = docId;
+            newSelectedDocs[index] = baseDocId;
+
+            setSelectedDocs(newSelectedDocs);
+          },
+        });
+      }
+
+      if (selectedDocs.length > 2) {
+        additional.push({
+          iconType: 'cross',
+          label: i18n.translate('unifiedDataTable.removeFromComparison', {
+            defaultMessage: 'Remove from comparison',
+          }),
+          size: 'xs',
+          onClick: () => {
+            setSelectedDocs(selectedDocs.filter((id) => id !== docId));
+          },
+        });
+      }
+
+      const columnDisplay = isPlainRecord
+        ? i18n.translate('unifiedDataTable.comparisonColumnResultDisplay', {
+            defaultMessage: 'Result {resultNumber}',
+            values: { resultNumber: Number(docId || 0) + 1 },
+          })
+        : doc.raw._id;
+
+      currentColumns.push({
+        id: docId,
+        display:
+          docIndex === 0 ? (
+            <EuiFlexGroup gutterSize="xs" alignItems="center" responsive={false}>
+              <EuiFlexItem grow={false}>
+                <EuiIcon type="pinFilled" />
```

Comment:
I actually tried to do this originally, but I ran into two issues:
1. We can't add it as tooltip on the icon itself since the column header is a button, so the nested focusable elements would break accessibility (and it messes with the data grid keyboard navigation).
2. The data grid automatically adds a default tooltip that can't be disabled based on the `displayAsText` prop, so the tooltips would conflict anyway: <img width="357" alt="current" src="https://github.com/elastic/kibana/assets/25592674/4e6e2eae-8a69-49a8-a961-898be501260a">

One option could be to set the `displayAsText` prop to something that better conveys the pinned state like this, although it's also used for aria labels so we'd still need it to be a label vs hint/description:
<img width="460" alt="ref" src="https://github.com/elastic/kibana/assets/25592674/c9dd76b8-5df5-42ab-b048-9b0c6150bde4">
<img width="354" alt="comp" src="https://github.com/elastic/kibana/assets/25592674/932970c2-7c5e-4ac1-b116-e097f8ad02f2">

@andreadelrio do you have any thoughts on what we could do here given the limitations?

Reply:
I like both options you are proposing. I do wonder if we should be including the word `document` in this somewhere? Some options that come to mind are: `Anchor document: QMVqp44BRlFr9ae08` and `Pinned document: QMVqp44BRlFr9ae08`. Maybe @amyjtechwriter can chime in.

Reply:
I think your solution is great here :D